testlib/src/test/java/com/diffplug/spotless/generic/LicenseHeaderStepTest.java

```diff
@@ -46,6 +54,26 @@ public void fromFile() throws Throwable {
 		assertOnResources(step, KEY_FILE_NOTAPPLIED, KEY_FILE_APPLIED);
 	}
 
+	@Test
+	public void should_apply_license_containing_YEAR_variable() throws Throwable {
+		FormatterStep step = LicenseHeaderStep.createFromFile(createTestFile(KEY_LICENSE_WITH_YEAR), StandardCharsets.UTF_8, LICENSE_HEADER_DELIMITER);
+
+		StepHarness.forStep(step)
+				.test(getTestResource(KEY_FILE_WITHOUT_LICENSE), getFileContentWithYEAR(currentYear()))
+				.testUnaffected(getFileContentWithYEAR(currentYear()))
+				.testUnaffected(getFileContentWithYEAR("2003"))
+				.testUnaffected(getFileContentWithYEAR("1990-2015"))
+				.test(getFileContentWithYEAR("not a year"), getFileContentWithYEAR(currentYear()));
```

Comment:
Unless I've misunderstood something, I think that this line should be like the following code snippet, so that we test that Spotless replaces `$YEAR` with the current year:

```java
.test(getFileContentWithYEAR("$YEAR"), getFileContentWithYEAR(currentYear()));
```

If `LicenseHeaderStep.java` currently replaces text like "not a year" with the current year, then it should absolutely be fixed to recognise "$YEAR" and "$YEAR" only.

Reply:
We want to check here that the regex `[0-9]{4}(-[0-9]{4})?` detect that the content in the current file is not in a good format. 
`not a year` is not in a good format,  so it should be replaced by a license having the current year.